samples/batch/transcription-enabled-storage/CreateTranscription/CreateTranscription.cs

```diff
@@ -0,0 +1,295 @@
+            var modelIds = new List<Guid>();
+
+            if (busMessage.EventType.Contains("BlobCreate", StringComparison.OrdinalIgnoreCase))
+            {
+                var audioFileUrl = await StorageUtilities.CreateSASAsync(CreateTranscriptionEnvironmentVariables.AzureWebJobsStorage, dataUri.AbsoluteUri, log).ConfigureAwait(false);
+                var containerName = StorageUtilities.GetContainerNameFromPath(dataUri.AbsoluteUri);
+                var audioFileName = StorageUtilities.GetFileNameFromPath(dataUri.AbsoluteUri);
+                var audioInputContainer = CreateTranscriptionEnvironmentVariables.AudioInputContainer;
+
+                if (StorageUtilities.IsItAudio(audioFileName) && !string.IsNullOrEmpty(containerName) && containerName.Equals(audioInputContainer, StringComparison.Ordinal))
+                {
+                    log.LogInformation($"Create request for audio file {audioFileName}");
+
+                    var jobName = StorageUtilities.GetFileNameWithoutExtension(audioFileName);
+
+                    var primaryLocale = CreateTranscriptionEnvironmentVariables.Locale;
+                    primaryLocale = primaryLocale.Split('|')[0].Trim();
+                    var locale = primaryLocale;
+
+                    string locationString;
+                    byte[] byteArray = null;
+
+                    try
+                    {
+                        var secondaryLocale = CreateTranscriptionEnvironmentVariables.SecondaryLocale;
+
+                        if (!string.IsNullOrEmpty(secondaryLocale) &&
+                            !secondaryLocale.Equals("None", StringComparison.OrdinalIgnoreCase) &&
+                            !secondaryLocale.Equals(primaryLocale, StringComparison.OrdinalIgnoreCase))
+                        {
+                            secondaryLocale = secondaryLocale.Split('|')[0].Trim();
+
+                            log.LogInformation($"Primary locale: {primaryLocale}");
+                            log.LogInformation($"Secondary locale: {secondaryLocale}");
+
+                            var languageID = new LanguageIdentification(subscriptionKey, subscriptionRegion);
+                            var fileExtension = Path.GetExtension(audioFileName);
+                            byteArray = await StorageUtilities.DownloadFileFromSAS(audioFileUrl).ConfigureAwait(false);
+                            locale = await languageID.DetectLanguage(byteArray, fileExtension, primaryLocale, secondaryLocale).ConfigureAwait(false);
+                            log.LogInformation($"Identified locale: {locale}");
+                        }
+
+                        modelIds = GetModelIds(locale.Equals(primaryLocale, StringComparison.OrdinalIgnoreCase), log);
+                        var properties = CreateTranscriptionPropertyBag();
+
+                        var client = new BatchClient(subscription.SubscriptionKey, subscription.LocationUri.AbsoluteUri, log);
```

Comment:
The code here looks very similar to another method above. Can we share the code, instead of duplication?

Reply:
yes, both call the same method now.